CONTRIBUTING.md

```diff
@@ -5,6 +5,16 @@
 OpenSearch is a community project that is built and maintained by people just like **you**.
 [This document](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md) explains how you can contribute to this and related projects.
 
+### Design Documents
+
+Sometimes a feature is large enough that a design document is required necessary.
+
+A feature is usually large enough to require a design document if it adds new Java classes or API endpoints.
```

Comment:
`If the feature adds new Java classes or API endpoints it is considered large enough to require a design document.`

Reply:
I would keep this fairly vague and mention that design documents are recommended for more _substantial_ features.